### PR TITLE
Updating outdated monitoring docs and manifest

### PR DESCRIPTION
**Pull request by kokhang**
The prometheus manifests are outdated and dont work with the latest kubernetes.
Upgrading it to work.

[skip ci]

### CODE REVIEW

File: cluster/examples/kubernetes/monitoring/prometheus.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ metadata:
   name: prometheus
   namespace: rook
 ---
-apiVersion: rbac.authorization.k8s.io/v1alpha1
+apiVersion: rbac.authorization.k8s.io/v1beta1
```
Comment (galexrt) [inline cluster/examples/kubernetes/monitoring/prometheus.yaml]: Shouldn't this be `v1` for 1.8?

Comment (kokhang) [reply cluster/examples/kubernetes/monitoring/prometheus.yaml]: yes but i am not sure if `v1` works for K8s 1.7.x or 1,6.x. I tried `v1beta1` on 1.8 and it worked. I can give it a try later.

**Comment by kokhang** [comment]
@galexrt `v1` doesnt work for k8s 1.6. I am getting the error:
```Error from server (BadRequest): error when creating "prometheus.yaml": ClusterRoleBinding in version "v1" cannot be handled as a ClusterRoleBinding: no kind "ClusterRoleBinding" is registered for version "rbac.authorization.k8s.io/v1"```

But leaving it as `v1beta1 ` does work for 1.8

**Comment by galexrt** [comment]
@kokhang I'm fine with going for `v1beta1` then.

**Comment by travisn** [comment]
@kokhang you manually tested on 1.6, 1.7, and 1.8? I don't believe we have integration tests for prometheus so we can't rely on the build.

**Comment by kokhang** [comment]
@travisn Tested on 1.7.x and 1.8.x and it works. For 1.6.x, it fails because that version of prometheus operator uses CRD and CRD is not available on 1.6. So i added a link to the older version of prometheus-operator for 1.6.x.

**Pull request merged**